test/test_distributions.py

```diff
@@ -230,81 +230,85 @@ def _is_batched_multivariate(jax_dist):
 def gen_values_within_bounds(constraint, size, key=random.PRNGKey(11)):
     eps = 1e-6
 
-    if isinstance(constraint, constraints._Boolean):
+    if constraint is constraints.boolean:
         return random.bernoulli(key, shape=size)
-    elif isinstance(constraint, constraints._GreaterThan):
+    elif isinstance(constraint, constraints.greater_than):
         return jnp.exp(random.normal(key, size)) + constraint.lower_bound + eps
-    elif isinstance(constraint, constraints._IntegerInterval):
+    elif isinstance(constraint, constraints.integer_interval):
         lower_bound = jnp.broadcast_to(constraint.lower_bound, size)
         upper_bound = jnp.broadcast_to(constraint.upper_bound, size)
         return random.randint(key, size, lower_bound, upper_bound + 1)
-    elif isinstance(constraint, constraints._IntegerGreaterThan):
+    elif isinstance(constraint, constraints.integer_greater_than):
         return constraint.lower_bound + random.poisson(key, np.array(5), shape=size)
-    elif isinstance(constraint, constraints._Interval):
+    elif isinstance(constraint, constraints.interval):
         lower_bound = jnp.broadcast_to(constraint.lower_bound, size)
         upper_bound = jnp.broadcast_to(constraint.upper_bound, size)
         return random.uniform(key, size, minval=lower_bound, maxval=upper_bound)
-    elif isinstance(constraint, (constraints._Real, constraints._RealVector)):
+    elif constraint in (constraints.real, constraints.real_vector):
```

Comment:
FYI - we may expose these classes in the API (https://github.com/pytorch/pytorch/issues/50616) so that this `ininstance/is` distinction can be avoided.